File: PhongTroAPI/PhongTroAPI/Entities/KhachThue.cs
using System;
using System.Collections.Generic;

namespace PhongTroAPI.Entities;

public partial class KhachThue
{
    public int MaKhach { get; set; }

    public string? HoTen { get; set; }

    public string? Sdt { get; set; }

    public string? Cccd { get; set; }

    public string? Email { get; set; }

    public DateOnly? NgaySinh { get; set; }

    public string? GioiTinh { get; set; }

    public string? DiaChiThuongTru { get; set; }

    public DateOnly? NgayCapCccd { get; set; }

    public string? NoiCapCccd { get; set; }

    public string? TrangThai { get; set; }

    public virtual ICollection<HoaDon> HoaDons { get; set; } = new List<HoaDon>();

    public virtual ICollection<HopDongThue> HopDongThues { get; set; } = new List<HopDongThue>();

    public virtual ICollection<LichHenXemPhong> LichHenXemPhongs { get; set; } = new List<LichHenXemPhong>();

    public virtual ICollection<Oghep> Ogheps { get; set; } = new List<Oghep>();

    public virtual ICollection<SuCo> SuCos { get; set; } = new List<SuCo>();

    public virtual ICollection<TaiKhoan> TaiKhoans { get; set; } = new List<TaiKhoan>();

    public virtual ICollection<ThongBao> ThongBaos { get; set; } = new List<ThongBao>();
}

File: trosmart/lib/models/khach_thue.dart
class KhachThue {
  final int maKhach;
  final String? hoTen;
  final String? sdt;
  final String? cccd;
  final String? email;
  final String? ngaySinh;
  final String? gioiTinh;
  final String? diaChiThuongTru;
  final String? ngayCapCccd;
  final String? noiCapCccd;
  final String? trangThai;

  KhachThue({
    required this.maKhach,
    this.hoTen,
    this.sdt,
    this.cccd,
    this.email,
    this.ngaySinh,
    this.gioiTinh,
    this.diaChiThuongTru,
    this.ngayCapCccd,
    this.noiCapCccd,
    this.trangThai,
  });

  factory KhachThue.fromJson(Map<String, dynamic> json) {
    return KhachThue(
      maKhach: json['maKhach'] ?? 0,
      hoTen: json['hoTen'],
      sdt: json['sdt'],
      cccd: json['cccd'],
      email: json['email'],
      ngaySinh: json['ngaySinh'],
      gioiTinh: json['gioiTinh'],
      diaChiThuongTru: json['diaChiThuongTru'],
      ngayCapCccd: json['ngayCapCccd'],
      noiCapCccd: json['noiCapCccd'],
      trangThai: json['trangThai'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'maKhach': maKhach,
      'hoTen': hoTen,
      'sdt': sdt,
      'cccd': cccd,
      'email': email,
      'ngaySinh': ngaySinh,
      'gioiTinh': gioiTinh,
      'diaChiThuongTru': diaChiThuongTru,
      'ngayCapCccd': ngayCapCccd,
      'noiCapCccd': noiCapCccd,
      'trangThai': trangThai,
    };
  }
}
